egcg_core/executor/cluster_executor.py

```diff
@@ -89,44 +92,56 @@ class PBSExecutor(ClusterExecutor):
     script_writer = script_writers.PBSWriter
 
     def _qstat(self):
-        h1, h2, data = self._get_stdout('qstat -x {j}'.format(j=self.job_id)).split('\n')
-        return data.split()
+        data = self._get_stdout('qstat -xt {j}'.format(j=self.job_id)).split('\n')
+        return [d for d in data[2:] if d]
 
-    def _job_status(self):
-        job_id, job_name, user, time, status, queue = self._qstat()
-        return status
+    def _job_statuses(self):
+        statuses = set()
+        reports = self._qstat()
+        for r in reports:
+            job_id, job_name, user, time, status, queue = r.split()
+            statuses.add(status)
+        return statuses
 
     def _job_exit_code(self):
-        return self.finished_statuses.index(self._job_status())
+        exit_status = 0
+        for s in self._job_statuses():
+            exit_status += self.finished_statuses.index(s)
+        return exit_status
 
 
 class SlurmExecutor(ClusterExecutor):
     unfinished_statuses = ('RUNNING', 'RESIZING', 'SUSPENDED', 'PENDING')
-    finished_statuses = ('COMPLETED', 'CANCELLED', 'FAILED', 'TIMEOUT', 'NODE_FAIL')
+    finished_statuses = ('COMPLETED', 'CANCELLED', 'CANCELLED+', 'FAILED', 'TIMEOUT', 'NODE_FAIL')
```

Comment:
There are other states that should be added to that list:
see [sacct docs](http://slurm.schedmd.com/sacct.html) in STATE section
Now that you raise an error in case of unknown status we need to make sure we have them all.